presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/PackageAPI.java

```diff
@@ -36,87 +36,96 @@
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
 import javax.ws.rs.QueryParam;
-import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 
 import java.net.MalformedURLException;
 import java.net.URL;
 
-import static com.google.common.io.Files.getFileExtension;
-import static javax.ws.rs.client.Entity.text;
+import static com.teradata.prestomanager.agent.PackageApiUtils.isRunning;
+import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
 import static javax.ws.rs.core.Response.Status.ACCEPTED;
 import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+import static javax.ws.rs.core.Response.Status.CONFLICT;
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
 
 @Path("/package")
 @Api(description = "the package API")
 @Singleton
+// TODO: Add configuration to get package type from PM config file
 public final class PackageAPI
 {
     private static final Logger LOGGER = LogManager.getLogger(PackageAPI.class);
-    private static PackageType installedPackageType;
 
     @PUT
-    @Consumes({MediaType.TEXT_PLAIN})
-    @Produces({MediaType.TEXT_PLAIN})
+    @Consumes({TEXT_PLAIN})
+    @Produces({TEXT_PLAIN})
     @ApiOperation(value = "Install presto using rpm or tarball")
     @ApiResponses(value = {
             @ApiResponse(code = 202, message = "Acknowledged request"),
-            @ApiResponse(code = 400, message = "Bad package format or Invalid url")
+            @ApiResponse(code = 400, message = "Invalid url"),
+            @ApiResponse(code = 409, message = "Presto is running")
     })
-    public synchronized Response install(@NotNull @ApiParam("Url to fetch package") String locationToFetchPackage,
+    public synchronized Response install(@ApiParam("Url to fetch package") String locationToFetchPackage,
             @QueryParam("checkDependencies") @DefaultValue("true") @ApiParam("If false, disables dependency checking") boolean checkDependencies)
     {
         LOGGER.debug("PUT /package  url: {} ; checkDependencies: {}", locationToFetchPackage, checkDependencies);
-        try {
-            installedPackageType = PackageType.valueOf(getFileExtension(locationToFetchPackage).toUpperCase());
-        }
-        catch (IllegalArgumentException e) {
-            LOGGER.error("Invalid package: {}. Package must be .tar.gz or .rpm", locationToFetchPackage, e);
-            return Response.status(BAD_REQUEST).entity(text("Invalid package type for presto-server. Package must be .tar.gz or .rpm")).build();
+        if ("".equals(locationToFetchPackage)) {
+            LOGGER.error("Url is empty or null");
+            return Response.status(BAD_REQUEST).entity("Expected URL in the body").build();
         }
         try {
+            if (isRunning()) {
```

Comment:
Its used to handle the case when user sends install command and the presto is already running.